From 4c13ed715cfd400911e903fcb3d46395cb293bdb Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Mon, 23 Aug 2021 10:48:27 +0200
Subject: [PATCH 16/18] riscv64: provide CHAR8 and CHAR16 (instead of WCHAR)

Shim has diverged from upstream gnu-efi in using CHAR16 (instead of WCHAR)
and CHAR8.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
---
 inc/riscv64/efibind.h | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/inc/riscv64/efibind.h b/inc/riscv64/efibind.h
index 0a818ae..a74a147 100644
--- a/inc/riscv64/efibind.h
+++ b/inc/riscv64/efibind.h
@@ -28,13 +28,11 @@ typedef int64_t                 INT64;
 typedef uint32_t                UINT32;
 typedef int32_t                 INT32;
 typedef uint16_t                UINT16;
+typedef uint16_t                CHAR16;
 typedef int16_t                 INT16;
 typedef uint8_t                 UINT8;
 typedef int8_t                  INT8;
-#ifndef __WCHAR_TYPE__
-#define __WCHAR_TYPE__          short
-#endif
-typedef __WCHAR_TYPE__          WCHAR;
+typedef char                    CHAR8;
 #ifndef BOOLEAN
 typedef uint8_t                 BOOLEAN;
 #endif
-- 
2.32.0

